docs/attach.rst

```diff
@@ -0,0 +1,90 @@
+Attaching to a running process
+==============================
+
+Memray allows you to attach to a running process and to observe the allocations
+it performs once you've attached. This doesn't allow you to see where memory
+was allocated before you attached to the process, but it does allow you to
+observe or record future allocations. This can be useful if an application is
+continuing to request more memory than you think it should need, and you want
+to figure out why. It can also be useful for observing the allocation patterns
+of a long running process once the process has already completed any initial
+warm up steps that it needs to do.
+
+Basic Usage
+-----------
+
+The general form of the ``attach`` subcommand is:
+
+.. code:: shell
+
+    memray attach [options] <pid>
+
+The only required argument for ``memray attach`` is a process ID to attempt to
+attach to. You must be :ref:`able to attach a debugger <ptrace privs>` to that
+process, and that process must be a Python process capable of running ``import
+memray`` (that is, the Memray package must be installed in the environment that
+the process is running in).
+
+By default this will open a :doc:`live mode TUI <live>` showing the allocations
+that the process performs after you've attached, but you can instead provide
+the name of a capture file to write to with the ``-o`` option, allowing you to
+analyze the captured allocations with any Memray reporter you'd like. You can
+also provide most of the options that ``memray run`` accepts. See :ref:`the
+CLI reference <memray attach CLI reference>` for details.
+
+.. _ptrace privs:
+
+Debugger Privileges
+-------------------
+
+Memray leverages a debugger for attaching to the process. It is compatible with
+both gdb and lldb, but one or the other must be installed in order for ``memray
+attach`` to work. Only a super user (either root, or a user with the
+``CAP_SYS_PTRACE`` capability) can attach to processes run by another user.
+Further, security settings on modern Linux systems typically prevent a regular
+user from attaching even to their own processes. You can loosen that
+restriction by writing ``0`` to ``/proc/sys/kernel/yama/ptrace_scope`` as root,
+allowing any user to attach to their own processes. When running a Docker
+container, you can use ``--cap-add=SYS_PTRACE`` to allow attaching to processes
+within the container.
```

Comment:
Let's add that in macOS the debugger may ask for a password or authentication before proceeding (this happens in my machine when I try to debug a process I own but that is not the child of the ataching process).